src/main/java/com/google/api/codegen/transformer/java/JavaMethodViewGenerator.java

```diff
@@ -128,30 +146,49 @@ public JavaMethodViewGenerator(SampleType sampleType) {
         }
         apiMethods.add(
             clientMethodTransformer.generateAsyncOperationRequestObjectMethod(
-                requestMethodContext));
+                requestMethodContext.withCallingForms(
+                    Collections.singletonList(CallingForm.LongRunningRequestAsync))));
+        apiMethods.add(
+            clientMethodTransformer.generateOperationCallableMethod(
+                requestMethodContext.withCallingForms(
+                    Collections.singletonList(CallingForm.LongRunningCallable))));
         apiMethods.add(
-            clientMethodTransformer.generateOperationCallableMethod(requestMethodContext));
-        apiMethods.add(clientMethodTransformer.generateCallableMethod(requestMethodContext));
+            clientMethodTransformer.generateCallableMethod(
+                requestMethodContext.withCallingForms(
+                    Collections.singletonList(CallingForm.Callable))));
       } else {
+
+        // Unary Methods.
```

Comment:
Have we tested samplegen works well for flattened methods (ie we only generate the flattened forms that match the configured params)? I filed an internal bug to check this.

Reply:
Flattening are configured by [`flattening`](https://github.com/googleapis/gapic-generator/blob/7a8780d4df66b3b44251243ba2f7bbf0444d5135/src/test/java/com/google/api/codegen/testsrc/common/library_gapic.yaml#L110) in gapic.yaml, so we can't actually make sure the generated samples always match the configured params (sample_value_set.parameters, IIUC) because that overload may not exist.

For Java I think we are generating samples for all flattening groups. C# samples are not yet ready for testing (mainly because import section is not handled yet).

That said, I think we need to be more explicit about the behavior of flattening methods. Let's chat tomorrow when I am in office.

Reply:
Yes, let's chat this week. I have some ideas, based on what we've been discussing offline.